include/live_sysallocations.hpp

```diff
@@ -0,0 +1,87 @@
+#pragma once
+
+#include "ddprof_defs.hpp"
+#include "logger.hpp"
+#include "unwind_output.hpp"
+
+#include <unordered_map>
+
+namespace ddprof {
+
+class SystemAllocation {
+private:
+  template <class T> T to_page(T a) {
+    return ((a + T{4095ull}) & (~T{4095ull})) >> T{12ull};
+  }
+
+public:
+  void add_allocs(const UnwindOutput &stack, uintptr_t addr, size_t size,
+                  pid_t pid) {
+    StackMap &stack_map = _pid_map[pid];
+
+    // Convert addr to page idx, then page-align size and decimate
+    uintptr_t page_start = to_page(addr);
+    uintptr_t page_end = to_page(addr + size);
+
+    for (auto i = page_start; i <= page_end; ++i) {
+      stack_map[i] = stack;
+    }
+  }
+
+  void move_allocs(uintptr_t addr0, uintptr_t addr1, size_t size, pid_t pid) {
```

Comment:
This is unused for now, but I'd like to keep it for a while since it isn't 100% clear to me that bulk movement (delete old + add new) is appropriate.